app/src/main/java/com/lagradost/cloudstream3/utils/BiometricAuthenticator.kt

```diff
@@ -0,0 +1,177 @@
+package com.lagradost.cloudstream3.utils
+
+import android.app.Activity
+import android.app.KeyguardManager
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import android.widget.Toast.LENGTH_SHORT
+import androidx.appcompat.app.AppCompatActivity
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_STRONG
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_WEAK
+import androidx.biometric.BiometricManager.Authenticators.DEVICE_CREDENTIAL
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import com.lagradost.cloudstream3.CommonActivity.showToast
+import com.lagradost.cloudstream3.MainActivity
+import com.lagradost.cloudstream3.R
+
+object BiometricAuthenticator {
+
+    private const val MAX_FAILED_ATTEMPTS = 3
+    private var failedAttempts = 0
+    const val TAG = "cs3Auth"
+
+    private lateinit var biometricManager: BiometricManager
+    lateinit var biometricPrompt: BiometricPrompt
```

Comment:
Please no lateinit vars, especially if they are public, its far too easy to shoot yourself in the foot using them.

Reply:
changed to nullable